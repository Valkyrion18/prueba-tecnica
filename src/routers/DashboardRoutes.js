import {
    Routes, Route,
    Navigate
} from 'react-router-dom';
import Home from '../components/Home';
import Products from '../components/Products/Products';

export const DashboardRoutes = () => {
    return (
        <>
            <div>
                {/* <NavBar /> */}
                <Routes>
                    <Route path="/" element={<Home />} />
                    <Route path="/products" element={<Products />} />
                    <Route path='*' element={<Navigate to="/" />} />
                </Routes>
            </div>

        </>
    )
}
