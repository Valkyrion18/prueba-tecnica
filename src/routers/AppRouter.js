import React,{useState,useEffect} from "react";
import {
  BrowserRouter as Router,
  Routes,
  Route,
} from "react-router-dom";
import Login from '../components/User/Login';
import { Register } from '../components/User/Register';
import { DashboardRoutes } from "./DashboardRoutes";
import { PrivateRoute } from "./PrivateRoute";
import { PublicRoute } from "./PublicRoute";
import { useDispatch } from 'react-redux';
import { getAuth,onAuthStateChanged } from "firebase/auth";
import { loginEmailPassword } from "../redux/actions/actionLogin";
// import styled from "styled-components";

// const H1 = styled.h1`
//     display: flex;
//     justify-content: center;
//     align-items: center;
//     color: black;
// `

export const AppRouter = () => {

  const dispatch = useDispatch()

  const [checking, setChecking] = useState(true)
  const [isLoggedIn, setIsLoggedIn] = useState(false)

  useEffect(() => {
    const auth = getAuth();
    onAuthStateChanged(auth, (user) => {
        if(user?.uid){
          dispatch(loginEmailPassword(user.uid, user.displayName));
         setIsLoggedIn(true)
        }
        else{
         setIsLoggedIn(false)
        }

        setChecking(false)
    })

 
}, [dispatch, setIsLoggedIn,setChecking])


if(checking){
  return(
      <h1>Un momento por favor...</h1>
  )
}

  return (
    <Router>
        <Routes>
         
        <Route path="/login" element={
                    <PublicRoute isAuthenticated={isLoggedIn}>
                        <Login/>                         
                    </PublicRoute>
                } />

                <Route path="/registro" element={
                    <PublicRoute isAuthenticated={isLoggedIn}>
                       <Register/>
                    </PublicRoute>
                } />

                
                <Route path="/*" element={
                    <PrivateRoute isAuthenticated={isLoggedIn}>
                        <DashboardRoutes/>
                    </PrivateRoute>
                }/>
        </Routes>
    </Router>
  );
}

