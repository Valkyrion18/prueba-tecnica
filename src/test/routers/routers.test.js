import React from "react";
import { shallow } from "enzyme";
import {PrivateRoute} from '../../routers/PrivateRoute';
import { PublicRoute } from "../../routers/PublicRoute";
import { DashboardRoutes } from "../../routers/DashboardRoutes";

describe ('Validar rutas Publicas y Privadas', () => {
    test('Validar el componente de Rutas Privadas', () => {
        const isLoggedIn = true;
        const wrapper = shallow(
            <PrivateRoute isAuthenticated={isLoggedIn} children={<DashboardRoutes />}/>)
        expect(wrapper).toMatchSnapshot();
    })
    test('Validar el componente de Rutas Publicas', () => {
        const isLoggedIn = true;
        const wrapper = shallow(
            <PublicRoute isAuthenticated={isLoggedIn} children={<DashboardRoutes />}/>)
        expect(wrapper).toMatchSnapshot();
    })
})