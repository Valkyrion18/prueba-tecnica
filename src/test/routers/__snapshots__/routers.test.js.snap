// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validar rutas Publicas y Privadas Validar el componente de Rutas Privadas 1`] = `<DashboardRoutes />`;

exports[`Validar rutas Publicas y Privadas Validar el componente de Rutas Publicas 1`] = `
<Navigate
  to="/*"
/>
`;
